### PR TITLE
[provider] s/AdoptInstances/UpdateController/

### PR DESCRIPTION
**Pull request by howbazaar**
Change the AdoptInstances environ method to be UpdateController as it needs to do more than just instances.

### CODE REVIEW

**Comment by howbazaar** [comment]
This change has been discussed with @mjs and @babbageclunk with respect to the providers tagging with the controller UUID.

File: environs/interface.go [hunk 1]
```diff
@@ -232,6 +232,15 @@ type Environ interface {
...
 	Create(CreateParams) error
 
+	// UpdateController is called when the model is moved from one controller
+	// to another using model migration. Some providers tag instances, disks,
+	// and cloud storage with the controller UUID to aid in clean destruction.
+	// All things that the provider has tagged with the controller UUID need
+	// to be updated in this method to use the new controller UUID. For
+	// providers that do not track the controller UUID, a simple method
+	// returning nil will suffice.
+	UpdateController(controllerUUID string) error
```
Comment (axw) [inline environs/interface.go]: Is it possible to include the agent version of the source controller? Two versions of a provider might tag things differently.

Comment (howbazaar) [inline environs/interface.go]: Hmm... interesting idea.

We were planning to run this on the source controller as part of the work in the migration master.
Perhaps now we should do it as part of the import on the target controller. We do need to consider how to roll back...

I'll wait and chat to @mjs and @babbageclunk tomorrow.

**Comment by anastasiamac** [comment]
This PR has conflicts.

**Comment by axw** [comment]
Pretty sure this is superseded, please reopen if not.

**Pull request closed without merging**